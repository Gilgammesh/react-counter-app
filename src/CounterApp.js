import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const [count, setCount] = useState(value);

  const handleAdd = () => {
    setCount(count + 1);
  };

  const handleReset = () => {
    setCount(value);
  };

  const handleSubtract = () => {
    setCount(count - 1);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{count}</h2>

      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleSubtract}>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  value: 123,
};

export default CounterApp;
