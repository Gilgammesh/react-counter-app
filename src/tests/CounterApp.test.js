import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import CounterApp from "../CounterApp";

describe("Pruebas unitarias en componente CounterApp.js", () => {
  /************************************************************************************/
  test("se debe mostrar <CounterApp /> correctamente", () => {
    const wrapper = shallow(<CounterApp />);

    expect(wrapper).toMatchSnapshot();
  });
  /************************************************************************************/
  test("se debe mostrar el valor por defecto de 100", () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);

    const countText = wrapper.find("h2").text().trim();

    expect(countText).toBe(`${value}`);
  });
  /************************************************************************************/
  test("se debe incrementar el value en 1 con el boton +1", () => {
    const value = 123;
    const wrapper = shallow(<CounterApp value={value} />);

    wrapper.find("button").at(0).simulate("click");

    const countText = wrapper.find("h2").text().trim();

    expect(countText).toBe(`${value + 1}`);
  });
  /************************************************************************************/
  test("se debe decrementar el value en 1 con el boton -1", () => {
    const value = 123;
    const wrapper = shallow(<CounterApp value={value} />);

    wrapper.find("button").at(2).simulate("click");

    const countText = wrapper.find("h2").text().trim();

    expect(countText).toBe(`${value - 1}`);
  });
  /************************************************************************************/
  test("se debe regresar al valor por defecto con el boton Reset", () => {
    const value = 105;
    const wrapper = shallow(<CounterApp value={105} />);

    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(1).simulate("click");
    const countText = wrapper.find("h2").text().trim();

    expect(countText).toBe(`${value}`);
  });
});
