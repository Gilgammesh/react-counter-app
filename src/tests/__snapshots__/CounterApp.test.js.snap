// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias en componente CounterApp.js se debe mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    123
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
